src/MudletServer.cpp

```diff
@@ -0,0 +1,131 @@
+{
+    mQueuedPackagePaths << packageName;
+}
+
+// Install the package queue on another instance of Mudlet.
+// Returns true on success
+bool MudletServer::installPackagesRemotely()
+{
+    // Pass the absolute path of the package to the active Mudlet Server
+    // The Mudlet Server may be owned by this process or another process.
+    QLocalSocket socket;
+    socket.connectToServer(mServerName);
+
+    if (socket.waitForConnected(WAIT_FOR_RESPONSE_MS)) {
+        const QString packagePathsData = mQueuedPackagePaths.join(QChar::LineFeed);
+        socket.write(packagePathsData.toUtf8());
+        socket.waitForBytesWritten(WAIT_FOR_RESPONSE_MS);
+        socket.disconnectFromServer();
+        return true;
+    }
+    return false;
+}
+
+
+// Attempt to start the Mudlet Server, return true if successful.
+// If a server was already running, cancel the attempt and return false.
+bool MudletServer::tryToStart()
+{
+    QLocalSocket socket;
+    socket.connectToServer(mServerName);
+
+    // If there is no response within a reasonable time:
+    // Assume no server exists, and start a new server.
+    if (!socket.waitForConnected(WAIT_FOR_RESPONSE_MS)) {
+        listen(mServerName);
+        return true;
+    }
+    return false;
+}
+
+// Install all queued packages to a specified profile/host.
+// Mudlet will call this function whenever a profile is activated.
+void MudletServer::installPackagesToHost(Host* activeProfile)
+{
+    mMutex.lock();
+    foreach(const QString& path, mQueuedPackagePaths) {
+        auto ret = activeProfile->installPackage(path, 0);
+    }
+    mQueuedPackagePaths.clear();
+    mMutex.unlock();
+}
+
+void MudletServer::incomingConnection(quintptr socketDescriptor)
+{
+    QLocalSocket* socket = new QLocalSocket(this);
+    socket->setSocketDescriptor(socketDescriptor);
+
+    connect(socket, &QLocalSocket::readyRead, this, &MudletServer::handleReadyRead);
+    connect(socket, &QLocalSocket::disconnected, this, &MudletServer::handleDisconnected);
+}
+
+// Receive package paths and install them
+void MudletServer::handleReadyRead()
+{
+    QLocalSocket* socket = qobject_cast<QLocalSocket*>(sender());
+    if (!socket) {
+        return;
+    }
+
+    mMutex.lock();
+    // Receive package paths and add them the install queue.
+    QByteArray data = socket->readAll();
+    const QStringList packagePaths = QString::fromUtf8(data).split(QChar::LineFeed, Qt::SkipEmptyParts);
+    mQueuedPackagePaths << packagePaths;
+    mMutex.unlock();
+
+    installPackagesLocally();
+}
+
+// Find the active host and install queued packages to it
+void MudletServer::installPackagesLocally()
+{
+    QTimer::singleShot(0, this, [this]() {
+        mudlet* mudletApp = mudlet::self();
+        Q_ASSERT(mudletApp);
+        Host* activeHost = mudletApp->getActiveHost();
+        if (activeHost) {
+            installPackagesToHost(activeHost);
+        } else {
+            mudletApp->slot_showConnectionDialog();
```

Comment:
Could you also add an 'info' notice to the connection dialog saying something along the lines of "choose a profile to install package(s) X into"? This way there would be some context as to why the connection dialog opened up on its own.

Reply:
![image](https://github.com/Mudlet/Mudlet/assets/147658676/020f6924-f394-47ac-877a-81a23da334e9)

Reply:
I think that looks good, and perhaps just the file name of the package is enough, not the entire path. 

In retrospect a message saying "Select and load a profile to install the following package(s) into:" would be clearer. Just choosing a profile without opening it won't do anything.